ambry-utils/src/test/java/com.github.ambry.utils/SimpleByteBufferPoolTest.java

```diff
@@ -215,13 +239,18 @@ public void testOneExpiredAnotherServed()
       used[i] = new CountDownLatch(1);
       consumers[i] = new BufferConsumer(size, blockTimeInMs[i], pool, allocated[i], used[i]);
       consumerThreads[i] = new Thread(consumers[i]);
-      consumerThreads[i].start();
     }
-    consumerThreads[1].join(); //wait until consumers[1] timeout
+    consumerThreads[0].start(); // Time t0
+    if (!allocated[0].await(100, TimeUnit.MILLISECONDS)) {
+      fail("SimpleByteBufferPool takes too long time to allocate a buffer to BufferConsumer[0].");
+    }
+    consumerThreads[1].start(); // Time t0 + delta
+    consumerThreads[2].start(); // Time t0 + delta
+    consumerThreads[1].join(); // wait until consumers[1] timeout
     if (!(consumers[1].exception instanceof TimeoutException)) {
       throw consumers[1].exception;
     }
-    used[0].countDown();
+    used[0].countDown(); // Time t2, R0 releases its buffer back to the buffer pool.
```

Comment:
usually we don't do same line comments. Since there are a lot in this file, we will leave it alone but just FYI